src/module/item/spell/overlay.ts

```diff
@@ -0,0 +1,137 @@
+import { PickAThingPrompt, PickAThingConstructorArgs, PickableThing } from "@module/apps/pick-a-thing-prompt";
+import { SpellOverlay, SpellOverlayType, SpellSource } from "./data";
+import { ErrorPF2e } from "@util";
+import { SpellPF2e } from ".";
+
+class SpellOverlayCollection extends Collection<SpellOverlay> {
+    constructor(public readonly spell: SpellPF2e, entries?: Record<string, SpellOverlay>) {
+        super(Object.entries(entries ?? {}));
+    }
+
+    /** Returns all variants based on override overlays */
+    get overrideVariants(): Embedded<SpellPF2e>[] {
+        return [...this.entries()].reduce((result: Embedded<SpellPF2e>[], [overlayId, data]) => {
+            if (data.overlayType === "override") {
+                const spell = this.spell.loadVariant({ overlayIds: [overlayId] });
+                if (spell) return [...result, spell];
+            }
+            return result;
+        }, []);
+    }
+
+    getType(overlayId: string): SpellOverlayType {
+        return this.get(overlayId, { strict: true }).overlayType;
+    }
+
+    async create(
+        overlayType: SpellOverlayType,
+        options: { renderSheet: boolean } = { renderSheet: false }
+    ): Promise<void> {
+        const id = randomID();
+
+        switch (overlayType) {
+            case "override":
+                await this.spell.update({
+                    [`data.overlays.${id}`]: {
+                        _id: id,
+                        sort: this.overrideVariants.length + 1,
+                        overlayType: "override",
+                    },
+                });
+                if (options.renderSheet) {
+                    const variantSpell = this.spell.loadVariant({ overlayIds: [id] });
+                    if (variantSpell) {
+                        variantSpell.sheet.render(true);
+                    }
+                }
+                break;
+        }
+    }
+
+    async update(overlayId: string, _data: Record<string, unknown>): Promise<void> {
```

Comment:
What is this method for? It doesn't seem to do anything.

Reply:
That will be the method to update the other types of overlay data. I was contemplating to make the first parameter `string | SpellPF2e` and integrating `updateOverrideData` into it but wasn't sure if that was a good idea.

Reply:
Might be better to not have this here at all for now, or if it needs to be stubbed for whatever reason, leave a comment.